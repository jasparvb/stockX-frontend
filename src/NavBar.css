.navbar-brand img {
  width: 140px;
}

.navbar .navbar-nav .nav-item a.nav-link {
  color: black;
}

.navbar .navbar-nav .nav-item a.nav-link:hover {
  color: rgb(0 200 5);
}