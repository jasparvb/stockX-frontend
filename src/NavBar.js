import React from "react";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";

function Navbar() {
    return (
        <nav className="navbar navbar-light bg-info">
        <Link to="/" className="navbar-brand text-light">
            <img src="%PUBLIC_URL%/stockX-logo.png"/>
        </Link>
        <ul className="navbar-nav flex-row">
            <li className="nav-item">
            <Link to="/login" className="nav-link text-light">
                Log In/Register
            </Link>
            </li>
        </ul>
        </nav>
    );
}

export default Navbar;
