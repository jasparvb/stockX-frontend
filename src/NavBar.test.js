import React from 'react';
import { render } from '@testing-library/react';
import NavBar from './NavBar';
import { Provider } from "react-redux";
import rootReducer from "./reducers/root";
import { createStore } from "redux";
import { BrowserRouter } from "react-router-dom";

const store = createStore(rootReducer);

it("renders without crashing", function() {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>);
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

