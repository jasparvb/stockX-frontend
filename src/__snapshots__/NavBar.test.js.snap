// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-light"
  >
    <a
      class="navbar-brand text-dark"
      href="/"
    >
      <img
        alt="stockX logo"
        src="stockX-logo.png"
      />
    </a>
    <div
      class="Search"
    >
      <form
        autocomplete="off"
        class="form-inline"
      >
        <input
          class="form-control form-control-lg flex-grow-1"
          name="query"
          placeholder="Search for stock or mutual fund"
          value=""
        />
        <div
          class="autocomplete "
        />
      </form>
    </div>
    <ul
      class="navbar-nav flex-row"
    >
      <li
        class="nav-item px-3"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Log In
        </a>
      </li>
      <li
        class="nav-item px-3"
      >
        <a
          class="btn btn-primary btn-md mr-4 mt-0"
          href="/login"
        >
          Sign Up
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
